modules/fieldlinessequence/rendering/renderablefieldlinessequence.cpp

```diff
@@ -345,206 +415,312 @@ RenderableFieldlinesSequence::RenderableFieldlinesSequence(
+        computeSequenceEndTime();
+    }
+    // Color group
+    _colorTablePath = FieldlinesSequenceModule::DefaultTransferFunctionFile.string();
     if (p.colorTablePaths.has_value()) {
-        _colorTablePaths = *p.colorTablePaths;
+        for (const std::filesystem::path& path : *p.colorTablePaths) {
+            if (!std::filesystem::exists(path)) {
+                throw ghoul::RuntimeError(std::format(
+                    "Color table path {} is not a valid file",
+                    path
+                ));
+            }
+            _colorTablePaths.emplace_back(path);
+        }
     }
-    else {
-        // Set a default color table, just in case the (optional) user defined paths are
-        // corrupt or not provided
-        _colorTablePaths.push_back(FieldlinesSequenceModule::DefaultTransferFunctionFile);
+    if (!p.colorTablePaths.has_value() || _colorTablePaths.empty()) {
+        _colorTablePaths.emplace_back(
+            FieldlinesSequenceModule::DefaultTransferFunctionFile
+        );
     }
-
     _colorUniform = p.color.value_or(_colorUniform);
-
     _colorMethod.addOption(static_cast<int>(ColorMethod::Uniform), "Uniform");
     _colorMethod.addOption(static_cast<int>(ColorMethod::ByQuantity), "By Quantity");
     if (p.colorMethod.has_value()) {
-        if (p.colorMethod.value() == "Uniform") {
-            _colorMethod = static_cast<int>(ColorMethod::Uniform);
-        }
-        else {
-            _colorMethod = static_cast<int>(ColorMethod::ByQuantity);
-        }
+        _colorMethod = static_cast<int>(
+            codegen::map<openspace::RenderableFieldlinesSequence::ColorMethod>(
+                *p.colorMethod
+            )
+        );
     }
     else {
         _colorMethod = static_cast<int>(ColorMethod::Uniform);
     }
-
-    if (p.colorQuantity.has_value()) {
-        _colorMethod = static_cast<int>(ColorMethod::ByQuantity);
-        _colorQuantityTemp = *p.colorQuantity;
-    }
+    _colorQuantityTemp = p.colorQuantity.value_or(_colorQuantityTemp);
 
     if (p.colorTableRanges.has_value()) {
         _colorTableRanges = *p.colorTableRanges;
     }
     else {
         _colorTableRanges.push_back(glm::vec2(0.f, 1.f));
+        _selectedColorRange = glm::vec2(0.f, 1.f);
     }
 
-    _loadingStatesDynamically = p.loadAtRuntime.value_or(_loadingStatesDynamically);
-    if (_loadingStatesDynamically && _inputFileType != SourceFileType::Osfls) {
-        LWARNING("Load at run time is only supported for osfls file type");
-        _loadingStatesDynamically = false;
+    if (p.colorMinMaxRange.has_value()) {
+        _selectedColorRange.setMinValue(glm::vec2(p.colorMinMaxRange->x));
+        _selectedColorRange.setMaxValue(glm::vec2(p.colorMinMaxRange->y));
     }
-
+    // To not change peoples masking settings i kept the parameter for the assets
+    // to be "MaskingRanges", but a single vec2-value instead of a vector.
+    // What is given from the asset, is stored as the selected range.
```

Comment:
This comment should be removed, IMO. It will quickly become outdated and is more a motivation for the changes in this PR rather than the code itself
```suggestion
```

If you keep it, add your name to it and the date, so that we know who "I" is and when this choice was made :)